python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -388,89 +404,293 @@ def _convolve(self, exposure: afwImage.Exposure, modelPsf: afwDetection.Gaussian
-                if sigma == "PsfFlux":
-                    # PSF photometry case
-                    aperSigma2 = targetSigma**2
-                    aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0)
-                    fluxScaling = 1.0  # Eq. A16 of Kuijken et al. (2015)
-                else:
-                    try:
-                        if sigma == "Optimal":  # Hook for DM-29290
-                            # Optimal elliptical aperture case
-                            # TODO: DM-29290
-                            # Set `normalize` to True to check if aperShape is valid.
-                            continue
-                        else:
-                            aperSigma2 = sigma**2 - targetSigma**2
-                            aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0,
-                                                           normalize=True)
-                            fluxScaling = 0.5*sigma**2/aperSigma2
-                    except (InvalidParameterError, ZeroDivisionError):
-                        # Raised when the aperture is invalid
-                        flagKey = measRecord.schema.join(baseName, "flag_bigpsf")
-                        measRecord.set(flagKey, 1)
-                        continue
-
-                fluxResult = measBase.SdssShapeAlgorithm.computeFixedMomentsFlux(convolved, aperShape, center)
-
-                # Calculate the integral in Eq. A17 of Kuijken et al. (2015)
-                # ``fluxErrScaling`` contains the factors not captured by
-                # ``fluxScaling`` and `instFluxErr`. It is 1 theoretically
-                # if ``kernelAcf`` is a Dirac-delta function.
-                fluxErrScaling = self._getFluxErrScaling(kernelAcf, aperShape)
-
-                # Scale the fluxResult and copy result to record
-                fluxResult.instFlux *= fluxScaling
-                fluxResult.instFluxErr *= fluxScaling*fluxErrScaling
-                fluxResultKey = FluxResultKey(measRecord.schema[baseName])
-                fluxResultKey.set(measRecord, fluxResult)
+            measureFlux = partial(self._measureFlux, measRecord, convolved, kernelAcf, center)
+
+            if self.config.doPsfPhotometry:
+                baseName = self.ConfigClass._getGaapResultName(scalingFactor, "PsfFlux", self.name)
+                aperShape = targetPsf.computeShape()
+                measureFlux(aperShape, baseName, fluxScaling=1)
+
+            # Iterate over pre-defined circular apertures
+            for sigma in self.config.sigmas:
+                baseName = self.ConfigClass._getGaapResultName(scalingFactor, sigma, self.name)
+                if sigma <= targetSigma:
+                    # Raise when the aperture is invalid
+                    self._setFlag(measRecord, baseName)
+                    continue
+
+                aperSigma2 = sigma**2 - targetSigma**2
+                aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0)
```

Comment:
As far as I can tell, before the change we had `normalize=True` passed on to `Quadrupole` here? Why this change?

Reply:
The `normalize=True` option is no longer required since we get here only if `aperSigma2` is positive. It wasn't necessarily the case previously (due to complex control flow) so I had implement the check here. Btw, `normalize=True` checks if the `Quadrupole` object has valid parameters and throws an `InvalidParameterError` if not.